cmd/kubeadm/app/master/manifests.go

```diff
@@ -304,7 +304,7 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted bool) [
 	}
 
 	command = append(getComponentBaseCommand(apiServer),
-		"--insecure-bind-address=127.0.0.1",
+		"--insecure-port=0",
```

Comment:
> Is it possible to get the "randomly selects a port" behavior? It would be useful for testing.

I think that belongs outside this layer.  See https://github.com/kubernetes/kubernetes/pull/42059/files#diff-a8c1e5100912cb27840594582aa0904bR66 as an example of how to wire it.